src/awesome_map/AwesomeMap.js

```diff
@@ -148,6 +148,18 @@ define(function(require) {
         this.onScaleChanged = Observable.newObservable();
 
         /**
+         * Observable for subscribing to pending scale changes.
+         *
+         * @method AwesomeMap#onScaleWillChange
+         * @param {Function} callback
+         *        Invoked with (sender, {
+         *            currentScale: {number}
+         *            nextScale: {number}
+         *        })
+         */
+        this.onScaleWillChange = Observable.newObservable();
```

Comment:
In my experience, most observables/events having to do with something that is going to happen usually allow the consumers to override what is going to happen. So can consumers change `nextScale`? Would we want to let them?